src/modules/audio/microphone.c

```diff
@@ -41,18 +41,22 @@ uint32_t lovrMicrophoneGetChannelCount(Microphone* microphone) {
   return microphone->channelCount;
 }
 
-SoundData* lovrMicrophoneGetData(Microphone* microphone) {
+SoundData* lovrMicrophoneGetData(Microphone* microphone, size_t samples) {
   if (!microphone->isRecording) {
     return NULL;
   }
 
-  size_t samples = lovrMicrophoneGetSampleCount(microphone);
-  if (samples == 0) {
+  size_t maxSamples = lovrMicrophoneGetSampleCount(microphone);
+  if (maxSamples == 0) {
     return NULL;
   }
+  lovrAssert(samples <= maxSamples, "Requested more audio data than is buffered by the microphone");
```

Comment:
Thoughts on using `MIN` here to clamp the input sample count to the max and make this function less throwy?  I don't have a strong opinion right now, but it's interesting to think about.

Reply:
It is, isn't it? I can't make up my mind. It's nice that the code is less error prone, but it's un-nice that it ends up doing something else than what was requested. If you lean slightly towards the latter being the lesser evil, I'll go with it.

Reply:
Yeah I think that's a good read.  I think it's the kind of thing that you just feel out while using the API for a while, so let's keep it in mind :)